share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md

```diff
@@ -0,0 +1,51 @@
+# How to Create and Maintain a Tap
+Taps are external sources of Homebrew formulae. They can be created by anyone
+to provide their own formulae to any Homebrew user.
+
+## Creating a tap
+A tap must be a git repository available online. It can be hosted on GitHub or
+elsewhere. If hosted on GitHub, we recommend that the repository’s name start
+with `homebrew-`.
+
+Formulae follow the same format as the core’s ones, and can be added at the
+repository’s root, or under `Formula` or `HomebrewFormula` subdirectories. We
+recommend the latter options because it makes the repository organisation
+easier to grasp, and top-level files are not mixed with formulae.
+
+See [homebrew-tex](https://github.com/Homebrew/homebrew-tex) for an example of
+a tap with a `Formula` subdirectory.
+
+### Installing
+If it’s on GitHub, users can install any of your formulae with
+`brew install user/repo/formula`. Homebrew will automatically add your
+`github.com/user/homebrew-repo` tap before installing the formula.
```

Comment:
Might be worth clarifying that `user/repo/formula` == `https://github.com/user/homebrew-repo/*/formula.rb`